std-bits/base/src/main/java/org/enso/base/Text_Utils.java

```diff
@@ -94,6 +96,8 @@ public static List<String> split_on_lines(String str, boolean keep_endings) {
       } else {
         currentPos += 1;
       }
+
+      context.safepoint();
```

Comment:
What's the impact on checking this on every single character?

Feels like this must have a significant detrimental affect?
On text processing like this every 10,000 characters would suffice or similar.

Reply:
There is a [Techniques and Applications for Guest-Language Safepoints paper](https://ssw.jku.at/Research/Papers/Daloze15.pdf) that claims that the guest language safe point check has zero performance overhead.

Reply:
That may be true, but this code is executed as _hosted_ Java code, not a _guest language_ code.